### DISCUSSION TITLE
failed reading zip archive: invalid header or archive is corrupted

### DISCUSSION THREAD

**Discussion by linyia01** [deepjavalibrary/djl#2761] [Development]
ai.djl.engine.EngineException: PytorchStreamReader failed reading zip archive: invalid header or archive is corrupted
	at ai.djl.pytorch.jni.PyTorchLibrary.moduleLoad(Native Method)
	at ai.djl.pytorch.jni.JniUtils.loadModuleHandle(JniUtils.java:1749)
	at ai.djl.pytorch.jni.JniUtils.loadModule(JniUtils.java:1732)
	at ai.djl.pytorch.engine.PtModel.load(PtModel.java:152)
	at ai.djl.pytorch.engine.PtModel.load(PtModel.java:139)
	at ai.djl.Model.load(Model.java:145)

<dependency>
            <groupId>ai.djl.pytorch</groupId>
            <artifactId>pytorch-engine</artifactId>
            <version>0.23.0</version>
        </dependency>
        <dependency>
            <groupId>ai.djl.pytorch</groupId>
            <artifactId>pytorch-native-cpu</artifactId>
            <version>1.13.1</version>
        </dependency>
        <dependency>
            <groupId>ai.djl.pytorch</groupId>
            <artifactId>pytorch-jni</artifactId>
            <version>1.13.1-0.23.0</version>
        </dependency>

    model = PriceNet(data_len1 - 2, data_len2 - data_len1).to(device)
    val_data =no.rand()
    item_data = val_data[0]
    x1 = torch.rand((1,data_len1 - 2))
    x2 = torch.rand((1,data_len2 - data_len1))

    model.eval()
    g = model(x1, x2)
    # traced_script_module=torch.jit.script(model)
    msave = torch.jit.script(model)
    # Save to file
    torch.jit.save(msave, "C:/wzry_model")
    # traced_script_module = torch.jit.trace(model, (x1, x2))
    # traced_script_module.save("C:/wzry_model.pt")

**Comment by frankfliu** [comment]
@linyia01 
Can you share your model and java code?

**Reply by linyia01** [reply]
python code is :
```
class PriceNet(nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, vec1, vec2):
        x1=torch.sum(vec1,dim=1)
        x2=torch.sum(vec2,dim=1)
        x = torch.add(x1, x2)
        return x

if __name__ == "__main__":
    device = torch.device("cpu")
    data_len1 = 1178
    data_len2 = 1243
    model = PriceNet()
    x1 = torch.rand((1,data_len1 - 2))
    x2 = torch.rand((1,data_len2 - data_len1))
    model.eval()
    g = model(x1, x2)
    traced_script_module = torch.jit.trace(model, (x1, x2))
    traced_script_module.save("C:/work/project/wzry2/model1")

    model = torchvision.models.resnet18(pretrained=True)
    model.eval()
    example = torch.rand(1, 3, 224, 224)
    traced_script_module = torch.jit.trace(model, example)
    traced_script_module.save("C:/work/project/wzry2/model2")
```
java code is :
```
            Model model = Model.newInstance("wzry_model", Device.cpu());
            ClassPathResource modelfile = new ClassPathResource("models/model1");
            InputStream inputStream = modelfile.getInputStream();
            System.out.println(inputStream.available());
            model.load(inputStream);
```

Looking forward to your answer  ,model1 is not ok  ,but model2 is ok

**Comment by linyia01** [comment]
> @linyia01 Can you share your model and java code?

**Reply by linyia01** [reply]
i have found zhe problem，when spring boot copy zhe file changed it
